### PR TITLE
Add wrapper structures, implement VM to template

### PR DESCRIPTION
**Pull request by tomsik68**
As we discussed, I've created wrapper structures for VM, IP and Storage that will provide ways of converting the GRPC objects into something that can be serialized to JSON, XML or as a template.

This PR also implements a wrapper object for VM that can be used to serialize VM as a template.

### CODE REVIEW

File: consumer/wrapper/storage_wrapper.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	return sw.st.GetRecordID()
+}
+
+func (sw storageWrapper) AsXML() interface{} {
+	// TODO
+	return storageXML{}
+}
+
+func (sw storageWrapper) AsJSON() interface{} {
+	return sw.st
```
Comment (amarthadan) [inline consumer/wrapper/storage_wrapper.go]: I think returning an empty structure would be better for methods that are useless for given data type. Or maybe even better some error message like `Not supported by given type` or something like that.

**Review by amarthadan — CHANGES_REQUESTED** [review]

File: consumer/wrapper/storage_wrapper.go (hunk 1 above)
Comment (tomsik68) [reply consumer/wrapper/storage_wrapper.go]: Yeah, I agree it makes a lot more sense to do just that. On the other hand, I'd have to change the return type to `interface{}, error`. I'd suggest to return a pointer to the structure and if the conversion is not implemented, return `nil`?

Comment (amarthadan) [reply consumer/wrapper/storage_wrapper.go]: Sorry, but I don't understand what you mean by that :smile:

**Review by amarthadan — APPROVED** [review]
Sorry, I didn't realize this is waiting for me.

**Pull request merged**